lib/Tools/LinearLayout.cpp

```diff
@@ -165,6 +165,35 @@ void assertDimsSubsetIgnoringOrder(T &&small, U &&big) {
   }
 }
 
+template <typename T, typename U>
+void assertCommonDimsSameOrder(T &&outerDims, U &&innerDims) {
+  // Check that elements common to both outerDimsRange and innerDimsRange
+  // appear in the same relative order.
+  SmallDenseSet<StringAttr> outerDimsSet(outerDims.begin(), outerDims.end());
+  SmallDenseSet<StringAttr> innerDimsSet(innerDims.begin(), innerDims.end());
+
+  std::vector<StringAttr> outerCommonDims;
+  for (StringAttr dim : outerDims) {
+    if (innerDimsSet.contains(dim)) {
+      outerCommonDims.push_back(dim);
+    }
+  }
+
+  std::vector<StringAttr> innerCommonDims;
+  for (StringAttr dim : innerDims) {
+    if (outerDimsSet.contains(dim)) {
+      innerCommonDims.push_back(dim);
+    }
+  }
+
+  if (outerCommonDims != innerCommonDims) {
+    llvm::report_fatal_error(
+        "Cannot multiply layouts.  All in/out dimensions common to both "
```

Comment:
Seems like this error is not correct anymore?

Reply:
Yeah, I think we can just remove "Cannot multiply layouts"